.github/workflows/basic-tests.yml

```diff
@@ -51,19 +51,20 @@ jobs:
         tox -e py-inference
     - name: store documentation page
       if: matrix.test-type == 'docs' && matrix.python-version == '3.8'
-      uses: actions/upload-artifact@v2
+      uses: actions/upload-artifact@v4
       with:
-        name: documentation-page
+        name: documentation-page-${{ matrix.os }}-{{ matrix.python-version }}
```

Comment:
This seems to be making/uploading the documentation for each OS and each python version to be deployed - is that right?

I'm not sure whether that is the model we are wanting

Reply:
> This seems to be making/uploading the documentation for each OS and each python version to be deployed - is that right?

not quite, in fact it merely enables it. I mindlessly applied the most general solution (which is to make artifact names unique) while I worked on upgrading a couple dozen repos at once, but I see that it's not actually needed here since only one os is present in the matrix and this step only runs on Python 3.8
I'll revert this one line.